week7/NYTArticles/app/src/main/java/com/example/nytarticles/App.kt

```diff
@@ -0,0 +1,35 @@
+package com.example.nytarticles
+
+import android.app.Application
+import androidx.room.Room
+import com.example.nytarticles.database.ArticleDB
+import com.example.nytarticles.database.ArticlesDAO
+import com.example.nytarticles.repository.ArticleRepo
+import com.example.nytarticles.networking.buildApiService
+
+class App : Application() {
+
+    companion object {
+        private lateinit var appInstance: App
+        fun getAppContext() = appInstance
+        private val articleDb: ArticleDB by lazy {
+            Room.databaseBuilder(appInstance, ArticleDB::class.java, DATABASE_NAME)
+                .build()
+        }
+        private val articlesDAO: ArticlesDAO by lazy {
+            articleDb.articleDao()
+        }
+        val BASE_URL = "https://api.nytimes.com/svc/topstories/v2/"
+        private val apiService by lazy { buildApiService() }
+        val articleRepo by lazy { ArticleRepo(apiService, articlesDAO) }
```

Comment:
Good job regarding the setup here!

So many people don't seem to understand they can do this, to reuse their dependencies! :]